test/abilities/sticky_hold.test.ts

```diff
@@ -0,0 +1,77 @@
+import { allMoves } from "#app/data/all-moves";
+import { StealHeldItemChanceAttr } from "#app/data/move-attrs/steal-held-item-chance-attr";
+import { Abilities } from "#enums/abilities";
+import { BerryType } from "#enums/berry-type";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import { GameManager } from "#test/testUtils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it, vi } from "vitest";
+
+describe("Abilities - Sticky Hold", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    // We are testing Sticky Hold on the enemy, since Knock Off's item removal is coded to not affect the player.
+    game.override
+      .ability(Abilities.BALL_FETCH)
+      .battleType("single")
+      .disableCrits()
+      .enemySpecies(Species.MAGIKARP)
+      .enemyAbility(Abilities.STICKY_HOLD)
+      .enemyHeldItems([{ name: "BERRY", type: BerryType.LUM }])
+      .enemyLevel(100);
+  });
+
+  it.each([
+    { moveName: "Thief", move: Moves.THIEF },
+    { moveName: "Pluck", move: Moves.PLUCK },
+    { moveName: "Incinerate", move: Moves.INCINERATE },
+    { moveName: "Knock Off", move: Moves.KNOCK_OFF },
+  ])("should prevent the user from losing a held item when hit by the move $moveName", async ({ move }) => {
```

Comment:
NIT: More flexible to potential changes to names:

```suggestion

  it.each([
    Moves.THIEF,
    Moves.PLUCK,
    Moves.INCINERATE,
    Moves.KNOCK_OFF,
  ].map(move => ({ move, name: Moves[move] })))("should prevent the user from losing a held item when hit by the move $name", async ({ move}) => {
```

----

A similar pattern could be applied in the following ones if desired